src/utils/infer-parser.js

```diff
@@ -55,10 +73,30 @@ function getLanguageByInterpreter(languages, file) {
   );
 }
 
+/**
+ * @param {SupportLanguage[]} languages
+ * @param {string | URL | undefined} [file]
+ * @returns {SupportLanguage | undefined}
+ */
+function getLanguageByIsSupported(languages, file) {
+  if (!file) {
+    return;
+  }
+
+  /*
+  We can't use `url.fileURLToPath` here since this module should work in browsers too
+  `URL#pathname` won't work either since `new URL('file:///C:/path/to/file').pathname`
+  equals to `/C:/path/to/file`, try to improve this part in future
+  */
+  file = String(file);
```

Comment:
Should we try to pass an absolute path here?

For your case, you want format `.husky/foo`, but if user runs prettier inside `.husky`, this won't work

```sh
prettier foo
```

Reply:
Yeah, I was assuming that `file` must be absolute here, but actually it could be relative path for example passed by user via `filepath` option with prettier's API.

Reply:
Alright, I checked that CLI format (including stdin) already use absolute path https://github.com/prettier/prettier/pull/14632, let's fix `--file-info`.

For APIs, user should always use absolute path to make it work.

Reply:
I'd like to raise another PR for changing `--file-info` later instead.

Reply:
Fixed in [`b0da113` (#17331)](https://github.com/prettier/prettier/pull/17331/commits/b0da1135bca9cbad883638dde718450857ae96ba)